docs/Tutorial_DiscordMERLIN.ipynb

```diff
@@ -0,0 +1,1356 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #81.                    min_dist = min_dist * (1 - warm_up_percentage * n_attempts)
</pre><p>Is it possible for <code>min_dist</code>  to be negative?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/TDAmeritrade/stumpy/pull/505/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Reply:
It can, but it will cause a problem if we want to calculate its corresponding dot-product value `r` in z-normalized Euclidean distance. 

Therefore, we better do: `min_dist = max(0, min_dist)`.  Thanks for catching that!

Then, if we already tried `min_dist=0` , and `n_attempts` is still less than `max_attempts`, we can break the loop. 

**NOTE**
if, similar to `decay`, we decide to give flexibility to the user, we can let them set a lower-bound value for `min_dist` if they want/need to. Let us call it `LB_min_dist`. then, we will have: `min_dist = max(LB_min_dist, min_dist)`

Reply:
> I'm really asking if the math will actually allow for the `min_dist` to be negative assuming that `decay` is always between `0` and `1`

Yes, that can happen. Notice that we are using the `ceil` in `max_attempts = int(np.ceil(1/decay[i])) `.  So, for instance, if `decay=0.6`, `max_attempts` becomes 2.  We will get negative value for `min_dist` in our second attempt using `min_dist = min_dist * (1 - decay[i] * n_attempts)`